src/legacy/core_plugins/kibana/server/lib/export/collect_references_deep.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { SavedObjectsClient } from '../../../../../../server/saved_objects/service/saved_objects_client';
+
+interface ObjectsToCollect {
+  id: string;
+  type: string;
+}
+
+export async function collectReferencesDeep(
+  savedObjectClient: SavedObjectsClient,
+  objects: ObjectsToCollect[]
+) {
+  const result = [];
+  const queue = [...objects];
+  while (queue.length !== 0) {
+    const itemsToGet = queue.splice(0, queue.length);
```

Comment:
Do you think we should limit the amount of items we fetch in each bulkGet? I bet it's highly unlikely to be important, but maybe just to be safe we should limit it to 1000 or something?